### PR TITLE
Embeds: Disable Customizer integration for block themes

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Make Customizer class conditional. This class is not needed in block based themes. 

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

See #9622 
See #9733

### CODE REVIEW

File: includes/Admin/Customizer.php [hunk 1]
```diff
@@ -124,6 +125,17 @@ public function register() {
...
 
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @since 1.16.0
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed(): bool {
+		return ! function_exists( 'wp_is_block_theme' ) || ! wp_is_block_theme();
```
Comment (swissspidy) [inline includes/Admin/Customizer.php:136]: FYI, this lint error reported by PHPStan is a false positive that should get resolved by #9973

File: includes/Admin/Customizer.php [hunk 2]
```diff
@@ -124,6 +125,21 @@ public function register() {
...
+	 * @since 1.16.0
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed(): bool {
+		if ( function_exists( '\wp_is_block_theme' ) ) {
+			return ! wp_is_block_theme();
+		}
+
+		return true;
```
Comment (swissspidy) [inline includes/Admin/Customizer.php]: What was wrong with `return ! function_exists( 'wp_is_block_theme' ) || ! wp_is_block_theme();`?

File: includes/Admin/Customizer.php [hunk 3]
```diff
@@ -124,6 +125,21 @@ public function register() {
...
 	}
 
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @since 1.16.0
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed(): bool {
```
Comment (swissspidy) [inline includes/Admin/Customizer.php:135]: Worth adding a test?

File: includes/Admin/Customizer.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Admin/Customizer.php]: Readability.

Comment (swissspidy) [reply includes/Admin/Customizer.php]: I found it quite readable 🤷

File: includes/Admin/Customizer.php (hunk 3 above)
Comment (spacedmonkey) [reply includes/Admin/Customizer.php:135]: What would it test? That is returns false?

Comment (swissspidy) [reply includes/Admin/Customizer.php:135]: I was thinking of a small unit test, but also an e2e test that verifies that the customizer is properly hidden.

File: includes/Admin/Customizer.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Admin/Customizer.php]: Changed back.

File: includes/Admin/Customizer.php (hunk 3 above)
Comment (spacedmonkey) [reply includes/Admin/Customizer.php:135]: How would I do that? I think that should be added once we test against 5.9? 
This change is simple and clear, no reason to block it.

Comment (spacedmonkey) [reply includes/Admin/Customizer.php:135]: Create a follow up ticket to add a e2e test?

Comment (swissspidy) [reply includes/Admin/Customizer.php:135]: > How would I do that?

Adding an e2e test would be pretty simple, no?

But yes, this needs to be done once we test against RC.

> This change is simple and clear, no reason to block it.

It's still a change that needs to be tested :-) Hence my original question whether it's worth adding tests.

**Review by swissspidy — APPROVED** [review]
Works as expected

**Pull request merged**